tests/kernel/queue/src/test_queue_contexts.c

```diff
@@ -28,10 +28,15 @@ static struct k_thread tdata;
 static K_THREAD_STACK_DEFINE(tstack1, STACK_SIZE);
 static struct k_thread tdata1;
 static struct k_sem end_sema;
+extern ZTEST_BMEM volatile bool valid_fault;
 
 void k_sys_fatal_error_handler(unsigned int reason, const z_arch_esf_t *pEsf)
 {
-	if (reason != K_ERR_KERNEL_OOPS) {
+	printk("Caught system error -- reason %d %d\n", reason, valid_fault);
```

Comment:
same question as in other PR....I remember reviewing a PR which adds a generic fatal error handling framework in tests, can we use that?

Reply:
HI @andrewboie , after a generic fatal error handling framework got merged, I have modified my code and re-push it, so could you take look at this PR again, thanks.